src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -165,10 +167,33 @@ bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler *
     if (!failSafeContext.IsFailSafeBusy() &&
         (!failSafeContext.IsFailSafeArmed() || failSafeContext.MatchesFabricIndex(accessingFabricIndex)))
     {
-        CheckSuccess(failSafeContext.ArmFailSafe(accessingFabricIndex, System::Clock::Seconds16(commandData.expiryLengthSeconds)),
-                     Failure);
-        response.errorCode = CommissioningError::kOk;
-        commandObj->AddResponse(commandPath, response);
+        // We do not allow CASE connections to arm the failsafe for the first time while the commissioning window is open in order
+        // to allow commissioners the opportunity to obtain this failsafe for the purpose of commissioning
+        if (!failSafeContext.IsFailSafeArmed() &&
+            Server::GetInstance().GetCommissioningWindowManager().CommissioningWindowStatus() !=
+                app::Clusters::AdministratorCommissioning::CommissioningWindowStatus::kWindowNotOpen &&
+            commandObj->GetExchangeContext()->GetSessionHandle()->GetSessionType() == Transport::Session::SessionType::kSecure &&
+            commandObj->GetExchangeContext()->GetSessionHandle()->AsSecureSession()->GetSecureSessionType() ==
+                Transport::SecureSession::Type::kCASE)
```

Comment:
Might be simpler as:
```suggestion
            commandObj->GetSubjectDescriptor().authMode == Access::AuthMode::kCase)
```
but either way.

Reply:
Added this manually in the next commit so I could test.